### PR TITLE
Feature/Conda

### PR DESCRIPTION
**Pull request by mblomdahl**
Sort-of same as #17 but for CentOS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,37 +15,32 @@ First setup [conda](https://conda.io/projects/conda/en/latest/) with Python 3.6,
...
-Creating the Conda environment files
-------------------------------------
+Creating the Conda Environment File
+-----------------------------------
 
 These commands needs to be executed on the target platform. The output file can then replace the
-corresponding file in this repo. I.E `environment-linux.yml` and `environment-macos.yml`,
+corresponding file in this repo, i.e. `environment-linux.yml` and `environment-macos.yml`:
 
-```bash
```
Comment (carlba) [inline README.md]: Are these commands not bash?

File: README.md [hunk 2]
```diff
@@ -15,37 +15,32 @@ First setup [conda](https://conda.io/projects/conda/en/latest/) with Python 3.6,
...
+    wget -P resources/ https://repo.continuum.io/archive/Anaconda3-2018.12-Linux-x86_64.sh
+    conda env create --name o3 python=3.6 -f environment-macos.yml
+    conda activate o3; pip install -e .
     export AIRFLOW_HOME=$(pwd)/airflow_home
     airflow initdb
     # Update newly-generated airflow.cfg in AIRFLOW_HOME by setting `dags_folder=$(pwd)/o3/dags`.
     airflow webserver -p 8080
     
     
-Creating the Conda environment files
```
Comment (carlba) [inline README.md]: There are two separate files albeit there are created in the same way. Both wordings were correct so why the change?

File: roles/airflow/templates/airflow-webserver.service.j2 [hunk 3]
```diff
@@ -7,7 +7,7 @@ EnvironmentFile=/home/airflow/airflow-environment
 User=airflow
 Group=airflow
 Type=simple
-ExecStart=/bin/bash -lc 'source /home/airflow/anaconda3/bin/activate o3; airflow webserver -p 8080'
+ExecStart=/bin/bash -l -c 'conda activate o3 && airflow webserver -p 8080'
```
Comment (carlba) [inline roles/airflow/templates/airflow-webserver.service.j2]: This does not work on Ubuntu did you test if it works on CentOS? 

```
● airflow-webserver.service - Airflow webserver daemon
   Loaded: loaded (/etc/systemd/system/airflow-webserver.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2019-01-29 20:59:13 UTC; 3s ago
  Process: 28228 ExecStart=/bin/bash -l -c conda activate o3 && airflow webserver -p 8080 (code=exited, status=127)
 Main PID: 28228 (code=exited, status=127)

Jan 29 20:59:13 ip-172-30-3-126 systemd[1]: Started Airflow webserver daemon.
Jan 29 20:59:13 ip-172-30-3-126 bash[28228]: /home/airflow/.profile: line 35: conda: command not found
Jan 29 20:59:13 ip-172-30-3-126 bash[28228]: /bin/bash: conda: command not found
```

I can assure you that `source /home/airflow/anaconda3/bin/activate o3` works on all environments.

File: roles/airflow/handlers/main.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 
 - name: airflow initdb
-  shell: bash -lc 'source /home/airflow/anaconda3/bin/activate o3 && airflow initdb'
+  shell: bash -l -c 'conda activate o3 && airflow initdb'
```
Comment (carlba) [inline roles/airflow/handlers/main.yml]: Does not work on Ubuntu it might work on CentOS I have not tested.

File: roles/airflow/tasks/main.yml [hunk 5]
```diff
@@ -62,75 +62,107 @@
...
+    - .conda
+    - .conda/pkgs
+  register: dot_conda_dirs
+  tags: airflow
+
+- name: create ~/.conda/environments.txt and ~/anaconda3/envs/.conda_envs_dir_test
+  file:
+    path: /home/airflow/{{ item }}
+    state: touch
+    mode: 0664
```
Comment (carlba) [inline roles/airflow/tasks/main.yml]: What about starting to use the much more readable way of doing this?

mode: "u=rw,g=rw,o=r"

File: roles/airflow/templates/airflow-webserver.service.j2 (hunk 3 above)
Comment (carlba) [reply roles/airflow/templates/airflow-webserver.service.j2]: This is solved by the `bash -ilc` fix

File: roles/airflow/handlers/main.yml (hunk 4 above)
Comment (carlba) [reply roles/airflow/handlers/main.yml]: This is solved by the `bash -ilc` fix

**Comment by carlba** [comment]
The merge conflict was my fault I committed the change in this branch. Probably should have made a separate PR for it. 

But if you check this commit this also works and is allowes the usage of `conda activate o3`:

https://github.com/mblomdahl/o3/pull/18/commits/353fa604a2f3ab1635b56c919ea4629dc0928154

File: README.md (hunk 1 above)
Comment (mblomdahl) [reply README.md]: Yes.

File: README.md (hunk 2 above)
Comment (mblomdahl) [reply README.md]: Correct.d
